### PR TITLE
Keybound omnislash

### PR DESCRIPTION
**Pull request by NegativeScript**

### CODE REVIEW

File: furusiya/model/keys/key_callbacks.py [hunk 1]
```diff
@@ -22,22 +25,22 @@ def enter_callback(event):
 # Movement
 @in_game(pass_turn=True)
 def up_callback(event):
-    Game.player.move_or_attack(0, -1)
+    up(Game.player.move_or_attack)
```
Comment (nightblade9) [inline furusiya/model/keys/key_callbacks.py]: I would call this (and the callback) `move_up`

File: tests/furusiya/model/skills/test_omnislash.py [hunk 2]
```diff
@@ -1,12 +1,52 @@
...
+    @pytest.fixture
+    def conf(self):
+        yield Mock(guaranteedHits=self.guaranteed_hits, rehitPercent=self.rehit_percent)
+
+    def test_slash_hits_guaranteed_hits(self, player, player_fighter, monster, monster_fighter, conf):
+        # Act
+        OmniSlash.process(player, monster, conf)
+
+        # Assert
+        assert player_fighter.attack.call_count >= self.guaranteed_hits
```
Comment (nightblade9) [inline tests/furusiya/model/skills/test_omnislash.py]: I think you can combine these two (or delete this test). Or alternatively, you can use a construct like "do this ten times and assert-fail if none of them are more than `guaranteed_hits`.

File: furusiya/model/skills/omnislash.py [hunk 3]
```diff
@@ -3,7 +3,19 @@
...
-        should_re_hit = Game.random.randint(0, 100) <= rehit_percent
-        if should_re_hit:
-            player.move_or_attack(*delta_coords)
+    def process(player, target, conf):
+        # do guaranteed hits
+        for _ in range(conf.guaranteedHits):
+            if Game.fighter_system.has(target):
+                Game.fighter_system.get(player).attack(target)
+            else:
+                return  # it's dead already!
```
Comment (nightblade9) [inline furusiya/model/skills/omnislash.py]: This could be a `break`, no? We're done if it's dead.

Comment (NegativeScript) [reply furusiya/model/skills/omnislash.py]: Yep. In this case, `return` works just as well. /shrug

File: tests/furusiya/model/skills/test_omnislash.py [hunk 4]
```diff
@@ -31,19 +31,23 @@ def monster_fighter(self, monster):
...
+            player_fighter.reset_mock()
 
     def test_slash_hits_extra_hits(self, player, player_fighter, monster, monster_fighter, conf):
         # set random seed for deterministic extra hits
         Game.random.seed(2)
-        conf.rehitPercent = 50
+        conf.probabilityOfAnotherHit = 50
 
         # Act
         OmniSlash.process(player, monster, conf)
```
Comment (nightblade9) [inline tests/furusiya/model/skills/test_omnislash.py]: The `+ 5` below is confusing. I would just change the equality check from `>=` to `>` and leave it at that.

**Pull request merged**